### PR TITLE
Convert tinderbox_scraper tests to pytest#468

### PR DESCRIPTION
**Pull request by Utkarsh1308**

### CODE REVIEW

File: tests/tinderbox_scraper/test_invalid_date_tinderbox.py [hunk 1]
```diff
@@ -6,36 +6,24 @@
...
-              'date': '2013-March-15',
-              'platform': 'win32'}}
-]
-
-tests = tests_with_invalid_date
+import pytest
 
 
+@pytest.mark.parametrize("args", [
+    ({'application': 'firefox', 'branch': 'mozilla-central', 'date': '20140317030202', 'locale': 'pt-PT', 'platform': 'win32'}),
```
Comment (whimboo) [inline tests/tinderbox_scraper/test_invalid_date_tinderbox.py]: nit: this line exceeds the 100 char line limit as shown by the linter check. Please wrap the line.

File: tests/tinderbox_scraper/test_invalid_date_tinderbox.py [hunk 2]
```diff
@@ -6,36 +6,24 @@
...
+
+def test_scraper(httpd, tmpdir, args):
+    """Testing download scenarios with invalid parameters for TinderboxScraper"""
+
+    with pytest.raises(ValueError):
+        TinderboxScraper(destination=str(tmpdir), base_url=httpd.get_url(), **args)
+
+
+'''
 class TestTinderboxScraperInvalidParameters(mhttpd.MozHttpdBaseTest):
```
Comment (whimboo) [inline tests/tinderbox_scraper/test_invalid_date_tinderbox.py]: You want to remove the old class.

File: tests/tinderbox_scraper/test_revision_tinderbox.py [hunk 3]
```diff
@@ -6,33 +6,30 @@
 
 from mock import patch
```
Comment (whimboo) [inline tests/tinderbox_scraper/test_revision_tinderbox.py]: Please change to make use of the pytest-mock module instead. See https://github.com/mozilla/mozdownload/blob/master/tests/try_scraper/test_invalid_revision.py how to do it.

**Comment by Utkarsh1308** [comment]
For some reason it's giving me an error for mocker not found in fixtures. I tried to run tox on the tests patched by rgtf as well and they are giving me the same error.

**Comment by Utkarsh1308** [comment]
@whimboo Please review

File: tests/tinderbox_scraper/test_revision_tinderbox.py [hunk 4]
```diff
@@ -4,35 +4,29 @@
...
-        with self.assertRaises(errors.NotFoundError):
-            TinderboxScraper(destination=self.temp_dir,
-                             base_url=self.wdir,
-                             logger=self.logger,
-                             platform='linux',
-                             revision='not_valid')
+
+def test_valid_revision(httpd, tmpdir, mocker):
+    query_builds_by_revision = mocker.patch('mozdownload.treeherder.Treeherder.query_builds_by_revision')
+    build_path = httpd.get_url() + '/firefox/tinderbox-builds/mozilla-central-linux/1374583608/'
```
Comment (whimboo) [inline tests/tinderbox_scraper/test_revision_tinderbox.py]: You have to check the error message by Travis CI, and you will immediately see what the problem actually is. The mocker works perfectly. But look yourself:

> E       AssertionError: assert 'http://127.0...-i686.tar.bz2' == 'http://127.0....-i686.tar.bz2'
E         - http://127.0.0.1:8080/firefox/tinderbox-builds/mozilla-central-linux/1374583608/firefox-25.0a1.en-US.linux-i686.tar.bz2
E         + http://127.0.0.1:8080//firefox/tinderbox-builds/mozilla-central-linux/1374583608/firefox-25.0a1.en-US.linux-i686.tar.bz2

**Comment by Utkarsh1308** [comment]
Please review @whimboo

File: tests/tinderbox_scraper/test_tinderbox_scraper.py [hunk 5]
```diff
@@ -7,12 +7,12 @@
...
+import pytest
+
 from mozdownload import TinderboxScraper
 from mozdownload.utils import urljoin
 
-import mozhttpd_base_test as mhttpd
-
-
+'''
 firefox_tests = [
```
Comment (whimboo) [inline tests/tinderbox_scraper/test_tinderbox_scraper.py]: Please remove all the old configs. The list of parametrized entries I will check then. Today I won't have the time for it anymore.

**Comment by Utkarsh1308** [comment]
@whimboo Please review

File: tests/tinderbox_scraper/test_invalid_date_tinderbox.py [hunk 6]
```diff
@@ -6,45 +6,19 @@
...
-              'date': '2013-March-15',
-              'platform': 'win32'}}
-]
+@pytest.mark.parametrize("args", [
+    ({'application': 'firefox', 'branch': 'mozilla-central', 'date': '20140317030202',
+    'locale': 'pt-PT', 'platform': 'win32'}),
+    ({'branch': 'mozilla-central', 'date': 'invalid', 'locale': 'pt-PT', 'platform': 'win32'}),
+    ({'branch': 'mozilla-central', 'date': '2013/07/02', 'platform': 'win64'}),
+    ({'branch': 'mozilla-central', 'date': '2013-March-15', 'platform': 'win32'}),
+])
```
[the remark this answers is not in the document]
Comment (Utkarsh1308) [reply tests/tinderbox_scraper/test_invalid_date_tinderbox.py]: In other tests I always had a line between the test function and the pytest args. Do I need to do this for the other 2 tests as well

[the remark this answers is not in the document]
Comment (whimboo) [reply tests/tinderbox_scraper/test_invalid_date_tinderbox.py]: Please do it for all the tests in that patch. Sorry, that it was an oversight from myself in former pull requests.

**Comment by Utkarsh1308** [comment]
Please review @whimboo

**Comment by whimboo** [comment]
This looks good now. Thanks a lot!

**Pull request merged**